File: cpp/55b_1490A_DenseArray.cpp
#include<bits/stdc++.h>

int count(int a, int b){
    int ret = 0;
    while((2*a) < b){
        ret++;
        a *= 2;
        //std::cout<<a<<" "<<ret<<"\n";
    }
    return ret;
}

int solve(std::vector<int> arr, int n){
    int cnt = 0;
    for(int i = 1; i < n; i++){
        if(2 * (std::min(arr[i], arr[i-1])) < (std::max(arr[i], arr[i-1]))){
            cnt += count(std::min(arr[i], arr[i-1]), std::max(arr[i],arr[i-1]));
        }
        //std::cout<<arr[i]<<" "<<cnt<<"\n";
    }
    return cnt;
}

int main(){
    int t, n, temp;
    std::vector<int> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>n;
        std::vector<int> arr;
        for(int j = 0; j < n; j++){
            std::cin>>temp;
            arr.push_back(temp);
        }
        ans.push_back(solve(arr, n));
    }
    for (auto i : ans){
        std::cout<<i<<'\n';
    }
    return 0;
}

File: python/55b_1490A_DenseArray.py
def count(a, b):
    ret = 0
    while 2 * a < b:
        ret += 1
        a *= 2
    return ret

def solve(arr, n):
    cnt = 0
    arr[0] = int(arr[0])
    for i in range(1, n):
        arr[i] = int(arr[i])
        if 2 * min(arr[i], arr[i-1]) < max(arr[i], arr[i-1]):
            cnt += count(min(arr[i], arr[i-1]), max(arr[i], arr[i-1]))
    return cnt

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        arr = input().split()
        ans.append(solve(arr, n))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()